bigquery_etl/backfill/shredder_mitigation.py

```diff
@@ -29,97 +47,260 @@ class ColumnStatus(Enum):
+        query += (
+            f" HAVING {having_clause}"
+            if having_clause is not None and group_by_clause is not None
+            else ""
+        )
+        query += f" ORDER BY {order_by_clause}" if order_by_clause is not None else ""
+        return query
 
+    def get_query_path_results(
+        self,
+        backfill_date: date = PREVIOUS_DATE,
+        row_limit: Optional[int] = None,
+        **kwargs,
+    ) -> list[dict[str, Any]]:
+        """Run the query in sql_path & return result or number of rows requested."""
+        having_clause = None
+        for key, value in kwargs.items():
+            if key.lower() == "having_clause":
+                having_clause = f"{value}"
+
+        with open(self.query_path, "r") as file:
+            sql_text = file.read().strip()
+
+        if sql_text.endswith(";"):
+            sql_text = sql_text[:-1]
+        if having_clause:
+            sql_text = f"{sql_text} {having_clause}"
+        if row_limit:
+            sql_text = f"{sql_text} LIMIT {row_limit}"
+
+        partition_field = self.partitioning["field"]
+        partition_type = (
+            "DATE" if self.partitioning["type"] == "DAY" else self.partitioning["type"]
+        )
+        parameters = None
+        if partition_field is not None:
+            parameters = [
+                bigquery.ScalarQueryParameter(
+                    partition_field, partition_type, backfill_date
+                ),
+            ]
 
-def get_bigquery_type(value) -> DataTypeGroup:
-    """Return the datatype of a value, grouping similar types."""
-    date_formats = [
-        ("%Y-%m-%d", date),
-        ("%Y-%m-%d %H:%M:%S", datetime),
-        ("%Y-%m-%dT%H:%M:%S", datetime),
-        ("%Y-%m-%dT%H:%M:%SZ", datetime),
-        ("%Y-%m-%d %H:%M:%S UTC", datetime),
-        ("%H:%M:%S", time),
-    ]
-    for format, dtype in date_formats:
         try:
-            if dtype == time:
-                parsed_to_time = datetime.strptime(value, format).time()
-                if isinstance(parsed_to_time, time):
-                    return DataTypeGroup.DATE
-            else:
-                parsed_to_date = datetime.strptime(value, format)
-                if isinstance(parsed_to_date, dtype):
-                    return DataTypeGroup.DATE
-        except (ValueError, TypeError):
-            continue
+            query_job = self.client.query(
+                query=sql_text,
+                job_config=bigquery.QueryJobConfig(
+                    query_parameters=parameters,
+                    use_legacy_sql=False,
+                    dry_run=False,
+                    use_query_cache=False,
+                ),
+            )
+            query_results = query_job.result()
+        except NotFound as e:
+            raise click.ClickException(
+                f"Unable to query current data, table {self.full_table_id} doesn't exist."
```

Comment:
here you might want to have the full name + the partition? (in case that the suffix)

Reply:
Yes, the partition is useful, I'll add it. The suffix (that I was using before) is removed.